lib/jsdom/living/helpers/dates-and-times.js

```diff
@@ -0,0 +1,245 @@
+"use strict";
+
+// TODO: use String.prototype.padStart instead when Node.js v8+ is required.
+const leftPad = require("left-pad");
+
+function isLeapYear(year) {
+  return year % 400 === 0 || (year % 4 === 0 && year % 100 !== 0);
+}
+
+// https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#number-of-days-in-month-month-of-year-year
+const daysInMonth = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+function numberOfDaysInMonthOfYear(month, year) {
+  if (month === 2 && isLeapYear(year)) {
+    return 29;
+  }
+  return daysInMonth[month - 1];
+}
+
+const monthRe = /^([0-9]{4,})-([0-9]{2})$/;
+
+// https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#parse-a-month-string
+function parseMonthString(str) {
+  const matches = monthRe.exec(str);
+  if (!matches) {
+    return null;
+  }
+  const year = Number(matches[1]);
+  if (year <= 0) {
+    return null;
+  }
+  const month = Number(matches[2]);
+  if (month < 1 || month > 12) {
+    return null;
+  }
+  return { year, month };
+}
+
+// https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#valid-month-string
+function isValidMonthString(str) {
+  return parseMonthString(str) !== null;
+}
+function serializeMonth({ year, month }) {
+  const yearStr = leftPad(`${year}`, 4, "0");
```

Comment:
We should drop the templating for these.

Reply:
This is needed to cast `year` to a string.